x-pack/plugins/ingest_manager/server/services/epm/agent/agent.ts

```diff
@@ -5,12 +5,71 @@
  */
 
 import Handlebars from 'handlebars';
+import { safeLoad } from 'js-yaml';
+import { DatasourceConfigRecord } from '../../../../common';
 
-interface StreamVars {
-  [k: string]: string | string[];
+function isValidKey(key: string) {
```

Comment:
Not a blocker for this PR but future improvement: I think these services belong in `/services/datasources.ts` too rather than EPM as EPM services are registry & asset interactions. Might be good to make a separate directory for data source services as that single file is getting rather long.